ui/src/app/machines/views/MachineDetails/MachineNetwork/AddInterface/AddInterface.tsx

```diff
@@ -0,0 +1,137 @@
+import { actions as machineActions } from "app/store/machine";
+import machineSelectors from "app/store/machine/selectors";
+import type { Machine, MachineDetails } from "app/store/machine/types";
+import {
+  NetworkInterfaceTypes,
+  NetworkLinkMode,
+} from "app/store/machine/types";
+import { getNextNicName } from "app/store/machine/utils";
+import type { RootState } from "app/store/root/types";
+import { actions as vlanActions } from "app/store/vlan";
+import vlanSelectors from "app/store/vlan/selectors";
+
+type Props = {
+  close: () => void;
+  systemId: MachineDetails["system_id"];
+};
+
+const InterfaceSchema = Yup.object().shape({
+  ip_address: Yup.string(),
+  mac_address: Yup.string()
+    .matches(MAC_ADDRESS_REGEX, "Invalid MAC address")
+    .required("MAC address is required"),
+  mode: Yup.mixed().oneOf(Object.values(NetworkLinkMode)),
+  name: Yup.string(),
+  fabric: Yup.number().required("Fabric is required"),
+  subnet: Yup.number(),
+  tags: Yup.array().of(Yup.string()),
+  vlan: Yup.number().required("VLAN is required"),
+});
+
+const AddInterface = ({ close, systemId }: Props): JSX.Element | null => {
+  const dispatch = useDispatch();
+  const machine = useSelector((state: RootState) =>
+    machineSelectors.getById(state, systemId)
+  );
+  const cleanup = useCallback(() => machineActions.cleanup(), []);
+  const errors = useSelector(machineSelectors.errors);
+  const fabrics = useSelector(fabricSelectors.all);
+  const fabricsLoaded = useSelector(fabricSelectors.loaded);
+  const creatingPhysical = useSelector((state: RootState) =>
+    machineSelectors.getStatusForMachine(state, systemId, "creatingPhysical")
+  );
+  const previousSaving = usePrevious(creatingPhysical);
+  const saved = !creatingPhysical && previousSaving && !errors;
+  const vlans = useSelector(vlanSelectors.all);
+  const vlansLoaded = useSelector(vlanSelectors.loaded);
+  const nextName = getNextNicName(machine, NetworkInterfaceTypes.PHYSICAL);
+
+  // Close the form if saving was successfully completed.
+  useEffect(() => {
+    if (saved) {
+      close();
+    }
+  }, [close, saved]);
+
+  useEffect(() => {
+    dispatch(fabricActions.fetch());
+    dispatch(vlanActions.fetch());
+  }, [dispatch]);
+
+  if (
+    !machine ||
+    !("interfaces" in machine) ||
+    !vlansLoaded ||
+    !fabricsLoaded
+  ) {
+    return <Spinner text="Loading..." />;
+  }
+  return (
+    <FormCard sidebar={false}>
+      <FormikForm
+        buttons={FormCardButtons}
+        cleanup={cleanup}
+        errors={errors}
+        initialValues={{
+          ip_address: "",
+          mac_address: "",
+          mode: NetworkLinkMode.LINK_UP,
+          name: nextName,
+          fabric: fabrics[0]?.id,
+          subnet: "",
+          tags: [],
+          vlan: vlans[0]?.id,
+        }}
+        onSaveAnalytics={{
+          action: "Add interface",
+          category: "Machine details networking",
+          label: "Add interface form",
```

Comment:
I think `label` is meant to be the actual text of the button, which in this case would be "Save interface"